pkg/registry/core/pod/storage/eviction.go

```diff
@@ -325,12 +326,22 @@ func addConditionAndDeletePod(r *EvictionREST, ctx context.Context, name string,
 		}
 
 		podCopyUpdated := rest.DefaultUpdatedObjectInfo(pod, conditionAppender)
+		podCopyUpdatedWithDeletePreconditions := &updatedObjectInfoWithOverridePreconditions{podCopyUpdated, deleteOptions.Preconditions}
 
-		if _, _, err = r.store.Update(ctx, name, podCopyUpdated, rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc, false, &metav1.UpdateOptions{}); err != nil {
+		updatedPodObject, _, err := r.store.Update(ctx, name, podCopyUpdatedWithDeletePreconditions, rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc, false, &metav1.UpdateOptions{DryRun: deleteOptions.DryRun})
+		if err != nil {
 			return err
 		}
+		if !resourceVersionIsUnset(deleteOptions) {
+			newResourceVersion, err := meta.NewAccessor().ResourceVersion(updatedPodObject)
+			if err != nil {
+				return err
+			}
+			// bump the resource version, since we are the one who modified it via the update
+			deleteOptions.Preconditions.ResourceVersion = &newResourceVersion
```

Comment:
mutating the input options is not something I'd recommend... work with a copy of deleteOptions and update it locally only.

if we need to propagate a resource version change to other invocations of this method (which I'm not sure we do), return it as a return value instead of mutating input variables

Reply:
+1, we do not need to propagate, I just overlooked this